### PR TITLE
v3 Client support for KubeControllersConfiguration

### PR DESCRIPTION
**Pull request by spikecurtis**
Signed-off-by: Spike Curtis <spike@tigera.io>

## Description

Adds support to the v3 client to read and write KubeControllersConfiguration resources.

## Todos
- [x] Tests
- [ ] Documentation
- [x] Release note

## Release Note

```release-note
v3 Client can CRUD KubeControllersConfiguration resources
```

### CODE REVIEW

File: lib/clientv3/kubecontrollersconfig_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+			Expect(res).To(MatchResourceWithStatus(apiv3.KindKubeControllersConfiguration, testutils.ExpectNoNamespace, name, spec2, status1))
+
+			By("Setting status2 on resource")
+			res.Status = status2
+			res, outError = c.KubeControllersConfiguration().Update(ctx, res, options.SetOptions{})
+			Expect(outError).ToNot(HaveOccurred())
+			Expect(res).To(MatchResourceWithStatus(apiv3.KindKubeControllersConfiguration, testutils.ExpectNoNamespace, name, spec2, status2))
+			rv1_3 := res.ResourceVersion
+
+			By("Getting resource and verifying status1 is present")
```
Comment (lmm) [inline lib/clientv3/kubecontrollersconfig_e2e_test.go]: Did you mean `status2`?

File: lib/clientv3/kubecontrollersconfig_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+				Spec:       spec1,
+			}, options.SetOptions{})
+			Expect(outError).To(HaveOccurred())
+			Expect(res).To(BeNil())
+			Expect(outError.Error()).To(Equal("error with field Metadata.UID = '' (field must be set for an Update request)"))
+
+			// Track the version of the updated name data.
+			rv1_2 := res1.ResourceVersion
+
+			By("Updating KubeControllersConfiguration name without specifying a resource version")
```
[the remark this answers is not in the document]
Comment (lmm) [reply lib/clientv3/kubecontrollersconfig_e2e_test.go]: Or maybe put `name` in parens like you do in other places?

[the remark this answers is not in the document]
Comment (spikecurtis) [reply lib/clientv3/kubecontrollersconfig_e2e_test.go]: I've dropped it. This was copied from another test, but in this case there is only one name and it has to be "default", so it's probably more clear without it.

File: lib/clientv3/kubecontrollersconfig_e2e_test.go (hunk 1 above)
Comment (spikecurtis) [reply lib/clientv3/kubecontrollersconfig_e2e_test.go]: Yep. Copy pasta

File: lib/backend/k8s/resources/kubecontrollersconfig.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	KubeControllersConfigCRDName      = "kubecontrollersconfigurations.crd.projectcalico.org"
+)
+
+func NewKubeControllersConfigClient(c *kubernetes.Clientset, r *rest.RESTClient) K8sResourceClient {
+	return &customK8sResourceClient{
+		clientSet:       c,
+		restClient:      r,
+		name:            KubeControllersConfigCRDName,
+		resource:        KubeControllersConfigResourceName,
+		description:     "Calico Cluster Information",
```
[the remark this answers is not in the document]
Comment (spikecurtis) [reply lib/backend/k8s/resources/kubecontrollersconfig.go]: Fixed

**Pull request merged**